### DISCUSSION TITLE
Problem with extra context when creating a response

### DISCUSSION THREAD

**Discussion by jannekanerva** [vitalik/django-ninja#1278] [Q&A]
This is simplified version of my endpoint. I have two Django models:

```
class Project(Model):
    name = ...

class Task(Model):
    name = ...
    project = ForeignKey(Project, ...)
```

Then I have a Django Ninja endpoint with the following schemas

```
class TaskSchema(ModelSchema):
    class Meta:
        model = Task
        fields = ('id', 'name', 'project')

class OutputSchema(Schema):
    tasks: List[Task]


@router.get('/', response=OutputSchema)
def tasks(request):
    tasks = Task.objects.select_related('Project')
    
    return {
        "tasks": tasks
    }
```

This works fine. The returned JSON is a list of Task objects where the project property is just an integer, just the way I want it to be. Like this:

```
{
  "tasks": [
    {
      "id": 1,
      "name": "Task",
      "project": 1
    },
    {...},
    {...},
  ]
}
```

However, I would like to pass extra context for the response serialization so that I can add a boolean property to each returned Task, let's say e.g. `is_favourite`.

I tried returning my response using `model_validate` method, but if I use that, I get a validation error.

```
def tasks(request):
    tasks = Task.objects.select_related('Project')
    favourite_ids = [1, 2, 3]  # Getting them from db for real
    
    return OutputSchema.model_validate({"tasks": tasks}, context={"favourite_ids": favourite_ids})
```

The error I get is the following, repeated for each task in the queryset.

```
response.tasks.0.project_id
  Field required [type=missing, input_value=<DjangoGetter: TaskSchema...other task', project=2)>, input_type=DjangoGetter]
```

So when I try passing the context, the previous validation starts to fail. I know I'm doing something wrong, but can't figure out what. All I want is to access the extra context and create an additional `is_favourite` field for each of my Tasks.

Any ideas how to make this work?

**Comment by manutheblacker** [comment]
Yoy may need to include your custom attributes into your schema using resolvers.
You can try this : https://django-ninja.dev/guides/response/?h=resolvers#resolvers

**Reply by jannekanerva** [reply]
> Yoy may need to include your custom attributes into your schema using resolvers.
> 
> You can try this : https://django-ninja.dev/guides/response/?h=resolvers#resolvers

The problem is I don't know how to pass the extra context for the resolver when returning a response. But I've come to the conclusion that it isn't possible in the current version - except adding a custom property to request object.

**Comment by vitalik** [comment]
@jannekanerva 

I think for the moment you can do it like this:

```Python
class TaskSchema(ModelSchema):
    is_favorite: bool
    class Meta:
        model = Task
        fields = ('id', 'name', 'project')

    @staticmethod
    def resolve_is_favorite(obj, context):
         request = context['request']
         return obj.id in request.context_favorites # <--------!

@router.get('/', response=OutputSchema)
def tasks(request):
    tasks = Task.objects.select_related('Project')
    request.context_favorites = [1, 2, 3] # <--------!
    
    return {
        "tasks": tasks
    }
```

or without context:

```Python
class TaskSchema(ModelSchema):
    is_favorite: bool
    class Meta:
        model = Task
        fields = ('id', 'name', 'project')


@router.get('/', response=OutputSchema)
def tasks(request):
    tasks = Task.objects.select_related('Project')
    favorites = [1, 2, 3] 
    tasks = list(tasks) # fetching from db # <--------!
    for t in tasks:
        t.is_favorite = t.id in favorites # <--------!
    
    return {
        "tasks": tasks
    }
```
